docs/meta.rst

```diff
@@ -17,55 +17,124 @@ Taichi kernels are *lazily instantiated* and a lot of computation can happen at
 Template metaprogramming
```

Comment:
I need to take a more careful pass on this part, which may take me a bit longer time. If you want to have this PR merged right now, could you move this file into a different PR? I'll take a look later. (Hopefully moving a single file isn't creating too much trouble for you... Sorry for being too busy these days...)

Reply:
Done, I'm super familiar with `git revert` these days, and may I merge this now?